src/components/Web3Status/index.tsx

```diff
@@ -141,6 +153,37 @@ function Web3StatusInner() {
 
   const { hasPendingActivity, pendingActivityCount } = usePendingActivity()
 
+  const displayName = useMemo(() => (account ? ENSName || shortenAddress(account) : null), [account, ENSName])
+  const meta = useRef(getConnectionMeta())
```

Comment:
using a ref for this makes the flow hard to follow. maybe if you wrapped it in a `useCurrentMeta` hook which handled updating when the connection loads, this file would be easier to follow